charts/redpanda/helpers.go

```diff
@@ -62,8 +62,46 @@ func FullLabels(dot *helmette.Dot) map[string]string {
 		"app.kubernetes.io/component":  Name(dot),
 	}
 
-	// TODO Shouldn't our defaults take precedence over user provided labels?
-	return helmette.Merge(defaults, labels)
+	return helmette.Merge(labels, defaults)
+}
+
+func StatefulSetPodLabelsSelector(dot *helmette.Dot, existingStatefulSetLabelSelector map[string]string) map[string]string {
+	if existingStatefulSetLabelSelector != nil {
```

Comment:
nit: Should we add in a check for `IsUpgrade` here, more so for clarity than anything else?

Reply:
Yes, I can add that.